score.go

```diff
@@ -127,14 +130,24 @@ func (d *indexData) scoreFileUsingBM25(fileMatch *FileMatch, doc uint32, cands [
 	termFreqs := map[string]int{}
 	for _, cand := range cands {
 		term := string(cand.substrLowered)
-		termFreqs[term]++
+
+		if cand.fileName {
+			termFreqs[term] += 2
+		} else {
+			termFreqs[term]++
+		}
 	}
 
 	// Compute the file length ratio. Usually the calculation would be based on terms, but using
 	// bytes should work fine, as we're just computing a ratio.
 	fileLength := float64(d.boundaries[doc+1] - d.boundaries[doc])
 	numFiles := len(d.boundaries)
```

Comment:
do we have to guard against `numFiles=0` too?

Reply:
I believe this should never happen -- if you had an empty shard, there's no possibility to match and score a doc?